static/certsplainer.js

```diff
@@ -0,0 +1,353 @@
+			"subject", "signatureAlgorithm", "keySize", "exponent", "curve",
+			"sha1hash", "sha256hash", "sha256_subject_spki", "pin-sha256", "id",
+			"permalink", "help"
+        ]) {
+        setField(id, "");
+    }
+    document.getElementById("curveRow").classList.remove("hidden");
+    document.getElementById("keySizeRow").classList.remove("hidden");
+    document.getElementById("exponentRow").classList.remove("hidden");
+}
+
+function clearExtensions() {
+    let extensionsTable = document.getElementById("extensions");
+    while (extensionsTable.children.length > 0) {
+        extensionsTable.children[0].remove();
+    }
+}
+
+function formatHTMLCommonName(name, id) {
+    return '<a href="/static/certsplainer.html?id=' + id + '">' + formatCommonName(name) + '</a>';
+}
+
+function formatCommonName(name) {
+    let result = "";
+    Object.keys(name).forEach((key) => {
+        if (key != "id") {
+            result += `/${key.toUpperCase()}=${name[key]}`;
+        }
+    });
+    return result;
+}
+
+function formatExtension(extensionName, extension) {
+    if (extensionName == "authorityKeyId" || extensionName == "subjectKeyId") {
+        let bin = atob(extension);
+        let hexStr = "";
+        for (let i = 0; i < bin.length; i++) {
+            hexStr += bin.charCodeAt(i).toString(16).padStart(2, "0");
+        }
+        return hexStr;
+    }
+
+    return extension.toString();
+}
+
+function setFieldsFromJSON(properties) {
+    clearFields();
+    clearExtensions();
+
+    if (!properties) {
+        return;
+    }
+
+    console.log(properties);
+
+    setField("version", properties.version);
+    setField("serialNumber", properties.serialNumber.toLowerCase());
+    setField("issuer", formatHTMLCommonName(properties.issuer, properties.issuer.id));
+    setField("notBefore", properties.validity.notBefore);
+    setField("notAfter", properties.validity.notAfter);
+    setField("subject", formatHTMLCommonName(properties.subject, properties.id));
+    setField("signatureAlgorithm", properties.signatureAlgorithm); // TODO technically there are two fields here...
+    if (properties.key.alg == "RSA") {
+        setField("keySize", properties.key.size);
+        setField("exponent", properties.key.exponent);
+        document.getElementById("curveRow").classList.add("hidden");
+    } else {
+        setField("curve", properties.key.curve);
+        document.getElementById("keySizeRow").classList.add("hidden");
+        document.getElementById("exponentRow").classList.add("hidden");
+    }
+    setField("sha1hash", properties.hashes.sha1.toLowerCase());
+    setField("sha256hash", properties.hashes.sha256.toLowerCase());
+    setField("sha256_subject_spki", properties.hashes.sha256_subject_spki.toLowerCase());
+    setField("pin-sha256", properties.hashes['pin-sha256'].toLowerCase());
+    setField("id", '<a href="/api/v1/certificate?id=' + properties.id + '">' + properties.id + '</a>');
```

Comment:
It'd be safer to use a templating library or push this into the HTML and show / hide / update it.